### PR TITLE
Reimplement MenuButton::switch_on_hover

### PR DESCRIPTION
**Pull request by guilhermefelipecgs**
Fix partially #37531. We still have some focus problems.

### CODE REVIEW

File: scene/gui/popup_menu.cpp [hunk 1]
```diff
@@ -1523,6 +1523,7 @@ PopupMenu::PopupMenu() {
 	set_hide_on_item_selection(true);
 	set_hide_on_checkable_item_selection(true);
 	set_hide_on_multistate_item_selection(false);
+	set_flag(Window::FLAG_NO_FOCUS, true);
```
Comment (guilhermefelipecgs) [inline scene/gui/popup_menu.cpp:1526]: This flag should theoretically fix the popup menu gaining focus, but sometimes it breaks for some reason.

Comment (guilhermefelipecgs) [reply scene/gui/popup_menu.cpp:1526]: I'll do some tests.

**Comment by pycbouh** [comment]
@YeldhamDev Do you think this is still relevant after your last two related PRs, https://github.com/godotengine/godot/pull/50619 and https://github.com/godotengine/godot/pull/50904?

**Comment by YeldhamDev** [comment]
Nope, not at all.

**Pull request closed without merging**